xarray/core/weighted.py

```diff
@@ -0,0 +1,307 @@
+from .computation import where, dot
+from typing import TYPE_CHECKING, Hashable, Iterable, Optional, Tuple, Union, overload
+
+if TYPE_CHECKING:
+    from .dataarray import DataArray, Dataset
+
+_WEIGHTED_REDUCE_DOCSTRING_TEMPLATE = """
+    Reduce this {cls}'s data by a weighted `{fcn}` along some dimension(s).
+
+    Parameters
+    ----------
+    dim : str or sequence of str, optional
+        Dimension(s) over which to apply the weighted `{fcn}`.
+    axis : int or sequence of int, optional
+        Axis(es) over which to apply the weighted `{fcn}`. Only one of the
+        'dim' and 'axis' arguments can be supplied. If neither are supplied,
+        then the weighted `{fcn}` is calculated over all axes.
```

Comment:
maybe remove as `xr.dot` does not provide this

Reply:
Agree re removing (or am I missing something—why was this here?)

Reply:
OK. Unless I'm missing something, let's remove `axis`